falcon/routing/converters.py

```diff
@@ -83,15 +83,50 @@ def convert(self, value):
         except ValueError:
             return None
 
+        return self._validate_min_max_value(value)
+
+    def _validate_min_max_value(self, value):
         if self._min is not None and value < self._min:
             return None
-
         if self._max is not None and value > self._max:
             return None
 
         return value
 
 
+class FloatConverter(IntConverter):
+    """Converts a field value to an float.
+
+    Identifier: `float`
+    Keyword Args:
+        min (float): Reject the value if it is less than this number.
+        max (float): Reject the value if it is greater than this number.
+    """
+
+    __slots__ = '_allow_nan'
+
+    def __init__(self, min=None, max=None, allow_nan=False):
+        self._min = min
+        self._max = max
+        self._allow_nan = allow_nan
+
+    def convert(self, value):
+        if not self._allow_nan:
+            if self._is_nan(value):
```

Comment:
I believe this check needs be after `value = float(value)` otherwise here it's comparing a string, so `value != value` is always false